thirdeye/thirdeye-frontend/app/pods/services/duration/service.js

```diff
@@ -0,0 +1,65 @@
+import Service from '@ember/service';
+import { assert } from '@ember/debug';
+import { isPresent } from "@ember/utils";
+
+export default Service.extend({
+  durationObj: null,
+  durationTypes: {
+    duration: 'string',
+    startDate: 'number',
+    endDate: 'number'
+  },
+
+  init() {
+    this._super(...arguments);
+    this.set('durationObj', {});
+  },
+
+  /**
+   * Saves new time range settings to persist
+   * @method setDuration
+   * @param {Object} newDuration - new incoming time range object
+   * @return {undefined}
+   */
+  setDuration(newDuration) {
+    const propsObj = this.get('durationTypes');
+    const requiredKeys = Object.keys(propsObj);
+    // Check each required property for presence and type
+    requiredKeys.forEach((key) => {
+      assert(`you must pass ${key} param as ${propsObj[key]}.`, typeof newDuration[key] === propsObj[key]);
+    });
+    this.set('durationObj', newDuration);
```

Comment:
I don't understand this loop. Why are we setting the same value `newDuration` to the `durationObj` object?

Reply:
Sure. `newDuration` is the incoming object containing the time-range properties to save.

It is not part of the loop. In the loop, I'm iterating over required keys to assert their presence and expected type.